### ISSUE TITLE
Prevent scheduled transaction to be sent in the same time by multiple nodes

### ISSUE THREAD

**Issue by samuel-uniris** [archethic-foundation/archethic-node#420]
### Is your feature request related to a problem?

We are using some self-triggered transactions based on scheduling like cronjob.
Hence, we should avoid two nodes to send the same transaction with few milliseconds of interval.
This can happen if the transaction is validated too fastly, creating some race condition.

### Describe the solution you'd like

The solution would be to either:
- store the timestamp of the last transaction in the scheduler state (coming from the replication) to not relaunch if timestamp > last scheduler.
- manage through FSM using event notifications

### Additional context

```mermaid
stateDiagram-v2
    [*] --> IDLE
    IDLE --> CONFIRMED: replication confirmation
    CONFIRMED --> IDLE: schedule new cron
    IDLE --> SCHEDULED: recv cron event
    SCHEDULED --> SCHEDULED: replication confirmation
    SCHEDULED --> CONFIRMED
 ```

**Comment by samuel-uniris** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Product-and-Design-Workspace-61efe71b9083a90018250d89/issues/archethic-foundation/archethic-node/420?planning-poker) with ZenHub @apoorv-2204 @imnik11 @Neylix @prix-uniris

**Comment by samuel-uniris** [comment]
Closed by #538
